File: backend/app/schemas/feeder.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class FeederBase(BaseModel):
    machine_id: int
    slot_number: int
    component_value: Optional[str] = None
    component_package: Optional[str] = None
    part_number: Optional[str] = None
    nozzle: int = 1
    pick_height: float = 0.0
    place_height: float = 0.0
    mount_speed: int = 100
    head: int = 0
    notes: Optional[str] = None


class FeederCreate(FeederBase):
    pass


class FeederUpdate(BaseModel):
    slot_number: Optional[int] = None
    component_value: Optional[str] = None
    component_package: Optional[str] = None
    part_number: Optional[str] = None
    nozzle: Optional[int] = None
    pick_height: Optional[float] = None
    place_height: Optional[float] = None
    mount_speed: Optional[int] = None
    head: Optional[int] = None
    notes: Optional[str] = None


class FeederResponse(FeederBase):
    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class FeederListResponse(BaseModel):
    items: list[FeederResponse]
    total: int

File: frontend/src/types/feeder.ts
export interface Feeder {
  id: number;
  machine_id: number;
  slot_number: number;
  component_value?: string;
  component_package?: string;
  part_number?: string;
  nozzle: number;
  pick_height: number;
  place_height: number;
  mount_speed: number;
  head: number;
  notes?: string;
  created_at?: string;
  updated_at?: string;
}

export interface FeederCreate {
  machine_id: number;
  slot_number: number;
  component_value?: string;
  component_package?: string;
  part_number?: string;
  nozzle?: number;
  pick_height?: number;
  place_height?: number;
  mount_speed?: number;
  head?: number;
  notes?: string;
}

export type FeederUpdate = Partial<Omit<FeederCreate, "machine_id">>;

export interface FeederListResponse {
  items: Feeder[];
  total: number;
}
